### PR TITLE
[SR] Align breadcrumbs with frontend

### PR DESCRIPTION
**Pull request by romtsn**
_#skip-changelog_

## :scroll: Description

* Change categories and data according to [this table](https://www.notion.so/sentry/Mobile-breadcrumb-UI-next-steps-8b6566484fd149259ffc5a05ecac7fe6?pvs=4#270369d36b1a4ad2b285fdd17a612e83) (what frontend expects)
* Had to introduce a new breadcrumb for battery status changes (with throttling of 1 minute, analogous to what iOS [does](https://github.com/getsentry/sentry-cocoa/blob/8aec30ebd6b60262f8b935b5475741c92da38834/Sources/Sentry/SentrySystemEventBreadcrumbs.m#L104))

## :bulb: Motivation and Context

Part of https://github.com/getsentry/sentry/issues/70065

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryLevel.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public void serialize(final @NotNull ObjectWriter writer, final @NotNull ILogger
     writer.value(name().toLowerCase(Locale.ROOT));
   }
 
-  static final class Deserializer implements JsonDeserializer<SentryLevel> {
+  public static final class Deserializer implements JsonDeserializer<SentryLevel> {
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/SentryLevel.java:21]: would it make sense to mark it as `@ApiStatus.Internal` then?

File: sentry-android-core/src/main/java/io/sentry/android/core/SystemEventsBreadcrumbsIntegration.java [hunk 2]
```diff
@@ -204,45 +207,69 @@ public void close() throws IOException {
...
-        for (String item : extras.keySet()) {
-          try {
-            @SuppressWarnings("deprecation")
-            Object value = extras.get(item);
-            if (value != null) {
-              newExtras.put(item, value.toString());
+      if (isBatteryChanged) {
+        final Float batteryLevel = DeviceInfoUtil.getBatteryLevel(intent, options);
+        if (batteryLevel != null) {
+          breadcrumb.setData("level", batteryLevel);
```
Comment (markushi) [inline sentry-android-core/src/main/java/io/sentry/android/core/SystemEventsBreadcrumbsIntegration.java:243]: float is fine here right? The notion doc is not 100% clear about this.

**Review by markushi — APPROVED** [review]
LGTM!

File: sentry-android-replay/src/main/java/io/sentry/android/replay/capture/BaseCaptureStrategy.kt [hunk 3]
```diff
@@ -201,12 +202,12 @@ internal abstract class BaseCaptureStrategy(
...
+                if (breadcrumb.timestamp.time >= segmentTimestamp.time &&
+                    breadcrumb.timestamp.time < endTimestamp.time
                 ) {
-                    // TODO: rework this later when aligned with iOS and frontend
                     var breadcrumbMessage: String? = null
-                    val breadcrumbCategory: String?
+                    var breadcrumbCategory: String? = null
+                    var breadcrumbLevel: SentryLevel? = null
                     val breadcrumbData = mutableMapOf<String, Any?>()
                     when {
```
Comment (vaind) [inline sentry-android-replay/src/main/java/io/sentry/android/replay/capture/BaseCaptureStrategy.kt:212]: @romtsn what do you think about extracting this to a class so we can override it from RN and Flutter to have custom mapping?

File: sentry/src/main/java/io/sentry/SentryLevel.java (hunk 1 above)
Comment (romtsn) [reply sentry/src/main/java/io/sentry/SentryLevel.java:21]: hm, we don't really do that for other Deserializers that are public, so I guess it's fine to keep them aligned?

**Pull request merged**